packages/babel-parser/src/plugin-utils.js

```diff
@@ -117,6 +117,15 @@ export function validatePlugins(plugins: PluginList) {
         RECORD_AND_TUPLE_SYNTAX_TYPES.map(p => `'${p}'`).join(", "),
     );
   }
+
+  if (
+    hasPlugin(plugins, "asyncDoExpressions") &&
+    !hasPlugin(plugins, "doExpressions")
+  ) {
+    throw new Error(
+      "'asyncDoExpressions' requires 'doExpressions', please add 'doExpressions' to parser plugins.",
+    );
```

Comment:
If we add `.missingPlugin = ["doExpressions"]` to the error (similarly to what `expectPlugin` does), `@babel/core` will suggest adding `plugin-proposal-do-expressions` to the config.